stdlib/LinearAlgebra/src/symmetric.jl

```diff
@@ -414,6 +414,45 @@ function triu(A::Symmetric, k::Integer=0)
     end
 end
 
+function dot(A::Symmetric{Ta,<:AbstractArray}, B::Symmetric{Tb,<:AbstractArray}) where {Ta,Tb<:Number}
+    n = size(A, 2)
+    if n != size(B, 2)
+        throw(DimensionMismatch("A has dimensions $(size(A)) but B has dimensions $(size(B))"))
+    end
+
+    dotprod = zero(dot(first(A), first(B)))
+    @inbounds if A.uplo == 'U' && B.uplo == 'U'
+        for j in 1:n
+            for i in 1:(j - 1)
+                dotprod += 2 * dot(A.data[i, j], B.data[i, j])
+            end
+            dotprod += dot(A.data[j, j], B.data[j, j])
+        end
+    elseif A.uplo == 'L' && B.uplo == 'L'
+        for j in 1:n
+            dotprod += dot(A.data[j, j], B.data[j, j])
+            for i in (j + 1):n
+                dotprod += 2 * dot(A.data[i, j], B.data[i, j])
+            end
+        end
+    elseif A.uplo == 'U' && B.uplo == 'L'
+        for j in 1:n
+            for i in 1:(j - 1)
+                dotprod += 2 * dot(A.data[i, j], B.data[j, i])
+            end
+            dotprod += dot(A.data[j, j], B.data[j, j])
+        end
+    elseif A.uplo == 'L' && B.uplo == 'U'
+        for j in 1:n
+            dotprod += dot(A.data[j, j], B.data[j, j])
+            for i in (j + 1):n
+                dotprod += 2 * dot(A.data[i, j], B.data[j, i])
```

Comment:
... some `transpose` here, but I’m not sure which (or if it makes a different at all whether at `A` or `B`).